### DISCUSSION TITLE
Crop image by polygon area

### DISCUSSION THREAD

**Discussion by akuzyagin** [deepjavalibrary/djl#2770] [Q&A] [answered]
Hello! How can I crop an image using a polygon shape? Needed in order to detect an object only in a specific area in the image.

Example code:
https://github.com/akuzyagin/video-detection-example/

eg, I think here to crop the image https://github.com/akuzyagin/video-detection-example/blob/2a92fe0a17c8c499f348fb1db8ace465c4d1e3b2/src/main/java/DetectionService.java#L45C1-L47C72:
```
Image image = ImageFactory.getInstance().fromImage(frame);
//TODO crop image by polygon before detect objects
DetectedObjects predict = predictor.predict(image);
```

![Screenshot_217](https://github.com/deepjavalibrary/djl/assets/127183480/0d8b7a5d-588d-4e99-a609-2f120fc316c1)

**Comment by frankfliu** [comment]
@akuzyagin 

An image must be a rectangle, what you can do is the cut the image to closet rectangle.

**Reply by akuzyagin** [reply]
Thanks for the answer! I found the answer to my question, how to cut out a polygon from an image. I can close this discussion because my solution seems closer. What do you think?

**Comment by akuzyagin** [comment, marked as answer]
I found the answer to my question, the following code allows you to cut out a polygon from an image:
![Screenshot_225](https://github.com/deepjavalibrary/djl/assets/127183480/b4e7318e-9251-47ce-973f-3d35bb2b2d1a)

```
// create polygon with coordinates for mask
MatOfPoint points = new MatOfPoint(
      new Point(100, 400),
      new Point(500, 350),
      new Point(700, 300),
      new Point(1000, 500),
      new Point(1000, 800),
      new Point(700, 800),
      new Point(500, 800),
      new Point(100, 400)
);

// draw shape with size of frame from original video and black fill
final Scalar colorBlack = new Scalar(0, 0, 0);
Mat mask = new Mat(frame.rows(), frame.cols(), CvType.CV_8UC3, colorBlack);

// fill polygon of white color on the black fill
final Scalar colorWhite = new Scalar(255, 255, 255);
Imgproc.fillPoly(mask, Collections.singletonList(points), colorWhite, Imgproc.LINE_8);

// compute the bitwise_and of the frame and mask
Mat result = new Mat();
Core.bitwise_and(frame, mask, result);

Image croppedImage = ImageFactory.getInstance().fromImage(result);
```

Full code: https://github.com/akuzyagin/video-detection-example/

**Comment by frankfliu** [comment]
This not really crop the image, it just mask the image. The whole image is still rectangle.

There are multiple ways to mask the image: see: https://pub.towardsai.net/how-to-identify-objects-at-pixel-level-using-deep-learning-in-java-907dc482835e
